guacamole-docker/bin/start.sh

```diff
@@ -536,6 +536,21 @@ END
 
 }
 
+##
+## Adds properties to guacamole.properties which configure the TOTP two-factor
+## authentication mechanism.
+##
+associate_totp() {
+    # Update config file
+    [ -n "$TOTP_ISSUER" ] && set_property "totp-issuer"    "$TOTP_ISSUER"
+    [ -n "$TOTP_DIGITS" ] && set_property "totp-digits"    "$TOTP_DIGITS"
+    [ -n "$TOTP_PERIOD" ] && set_property "totp-period"    "$TOTP_PERIOD"
+    [ -n "$TOTP_MODE" ]   && set_property "totp-mode"      "$TOTP_MODE"
```

Comment:
Looking at the other sections (mostly above this), it looks like instead of doing
```
 [ -n "$TOTP_ISSUER" ] && set_property "totp-issuer"    "$TOTP_ISSUER"
```
you should be able to do:
```
set_optional_property "totp-issuer" "$TOTP_ISSUER"
```

Reply:
I'm not sure because we would then set empty options in the configuration file.
In the other sections, options are tested to always have a value.

Reply:
This is the code for `set_optional_property()`:
```
set_optional_property() {
    NAME="$1"
    VALUE="$2"
    # Set the property only if a value is provided
    if [ -n "$VALUE" ]; then
        set_property "$NAME" "$VALUE"
    fi
}
```

So, it's essentially duplicating what you're doing, no?

Reply:
Tired certainly...
Thank you @necouchman for pointing this, just fixed 👍